server/bleep/src/webserver/answer.rs

```diff
@@ -44,6 +44,30 @@ use response::{Exchange, SearchResult, SearchStep, Update};
 
 const TIMEOUT_SECS: u64 = 60;
 
+#[derive(Clone, Debug, serde::Deserialize)]
+pub struct Vote {
+    pub positive: bool,
+    pub thread_id: uuid::Uuid,
+    pub query_id: uuid::Uuid,
+    pub repo_ref: Option<RepoRef>,
+}
+
+pub(super) async fn vote(
+    Query(params): Query<Vote>,
```

Comment:
we shouldn't use query params for POST requests, we should parse request body instead